rapids-cmake/cuda/set_runtime.cmake

```diff
@@ -0,0 +1,65 @@
+#=============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cuda_set_runtime
+-------------------------------
+
+.. versionadded:: v23.08.00
+
+Establish what CUDA runtime library should be used by a single target
+
+  .. code-block:: cmake
+
+    rapids_cuda_set_runtime( target USE_STATIC (TRUE|FALSE) )
```

Comment:
Rather than having this be a true/false, should we instead have `USE_STATIC` be an option rather than a one-valued parameter?

Also, if we want to encourage static linking of cudart perhaps we should make it an option `USE_SHARED` and otherwise default to use static. What do you think?

Reply:
The current mode of having it be `USE_STATIC` was selected so that it reduced the amount of calling code since RAPIDS projects generally have an option to control this called `CUDA_STATIC_RUNTIME`. 

So people can do:
```cmake
rapids_cuda_set_runtime(<target> USE_STATIC ${CUDA_STATIC_RUNTIME})
```

instead of:
```cmake
if(CUDA_STATIC_RUNTIME)
  rapids_cuda_set_runtime(<target> USE_STATIC)
else()
  rapids_cuda_set_runtime(<target>)
endif()
```


Your suggestion of having a default value of `TRUE` does make sense though.

Reply:
Makes sense, reducing friction here is probably more valuable than purity.